src/lex.cpp

```diff
@@ -1337,6 +1339,75 @@ void lexer::parse_number() {
     QLJS_SLOW_ASSERT(
         !(number_begin[0] == u8'0' && this->is_digit(number_begin[1])));
   }
+  if (!has_decimal_point && !has_exponent && !is_bigint) {
+    std::string cleaned_string = "";
+    for (char8 c : string8_view(number_begin, input - number_begin)) {
+      if (c != '_') {
+        cleaned_string.push_back(static_cast<char>(c));
+      }
+    }
+    if (cleaned_string.size() > 15) {
```

Comment:
Minor: Can you put the magic number 15 in a constant, and document that constant? I don't know what's special about 15.

Reply:
Every integer with 15 or fewer digits can be accurately represented without any loss of precision.